shared/src/library/kotlin/org/jetbrains/kotlin/konan/library/SearchPathResolver.kt

```diff
@@ -6,48 +6,32 @@ import org.jetbrains.kotlin.konan.library.impl.KonanLibraryImpl
 import org.jetbrains.kotlin.konan.library.impl.createKonanLibrary
 import org.jetbrains.kotlin.konan.target.Distribution
 import org.jetbrains.kotlin.konan.target.KonanTarget
-import org.jetbrains.kotlin.konan.util.*
-import kotlin.system.exitProcess
-import org.jetbrains.kotlin.library.KLIB_FILE_EXTENSION_WITH_DOT
-import org.jetbrains.kotlin.library.KotlinAbiVersion
-import org.jetbrains.kotlin.library.UnresolvedLibrary
-import org.jetbrains.kotlin.library.uniqueName
+import org.jetbrains.kotlin.library.*
+import org.jetbrains.kotlin.util.DummyLogger
+import org.jetbrains.kotlin.util.Logger
 
-// FIXME(ddol): KLIB-REFACTORING-CLEANUP: remove the constants below:
 const val KONAN_STDLIB_NAME = "stdlib"
 
-// FIXME(ddol): KLIB-REFACTORING-CLEANUP: remove this interface!
-interface SearchPathResolver : WithLogger {
-    val searchRoots: List<File>
-    fun resolutionSequence(givenPath: String): Sequence<File>
-    fun resolve(unresolved: UnresolvedLibrary, isDefaultLink: Boolean = false): KonanLibraryImpl
-    fun resolve(givenPath: String): KonanLibraryImpl
-    fun defaultLinks(noStdLib: Boolean, noDefaultLibs: Boolean, noEndorsedLibs: Boolean): List<KonanLibraryImpl>
-}
-
-// FIXME(ddol): KLIB-REFACTORING-CLEANUP: remove this interface!
-interface SearchPathResolverWithTarget: SearchPathResolver {
+interface SearchPathResolverWithTarget<out L: KotlinLibrary>: SearchPathResolverWithAttributes<L> {
     val target: KonanTarget
-    val knownAbiVersions: List<KotlinAbiVersion>?
-    val knownCompilerVersions: List<KonanVersion>?
 }
 
 fun defaultResolver(
         repositories: List<String>,
         target: KonanTarget,
         distribution: Distribution = Distribution(),
         compatibleCompilerVersions: List<KonanVersion> = emptyList()
-): SearchPathResolverWithTarget = defaultResolver(repositories, emptyList(), target, distribution, compatibleCompilerVersions)
+): SearchPathResolverWithTarget<KonanLibraryImpl> = defaultResolver(repositories, emptyList(), target, distribution, compatibleCompilerVersions)
 
 fun defaultResolver(
-        repositories: List<String>,
-        directLibs: List<String>,
-        target: KonanTarget,
-        distribution: Distribution,
-        compatibleCompilerVersions: List<KonanVersion> = emptyList(),
-        logger: Logger = DummyLogger,
-        skipCurrentDir: Boolean = false
-): SearchPathResolverWithTarget = KonanLibraryProperResolver(
+    repositories: List<String>,
+    directLibs: List<String>,
+    target: KonanTarget,
+    distribution: Distribution,
+    compatibleCompilerVersions: List<KonanVersion> = emptyList(),
+    logger: Logger = DummyLogger,
```

Comment:
BTW, shall it be indeed default logger?

Reply:
This is just a default argument value. It allows klib infrastructure to be used in the absence of  Kotlin compiler infrastructure (the message collector, etc). For the compiler we use an object based on the message collector